packages/migrate/src/commands/DbPull.ts

```diff
@@ -163,41 +163,47 @@ Set composite types introspection depth to 2 levels
      *
      * If neither these variables were set, we'd have already thrown a `NoSchemaFoundError`.
      */
-    const schema = await match({ url, schemaPath })
+    const { config, schema } = await match({ url, schemaPath })
       .when(
         (input): input is { url: string | undefined; schemaPath: string } => input.schemaPath !== null,
         async (input) => {
           const rawSchema = fs.readFileSync(input.schemaPath, 'utf-8')
+          const config = await getConfig({
+            datamodel: rawSchema,
+            ignoreEnvVarErrors: true,
+          })
 
           if (input.url) {
-            const config = await getConfig({
-              datamodel: rawSchema,
-              ignoreEnvVarErrors: true,
-            })
+            // TODO: ensure that the `input.url` protocol is compatible with the schema provider
+            // to avoid ambiguities in case of schema parsing errors (error code P1012)
```

Comment:
This would require https://github.com/prisma/prisma/issues/16201 to be solved before proceeding